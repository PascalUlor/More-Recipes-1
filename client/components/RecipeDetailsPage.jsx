import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
/** ***************************** RECIPE DETAILS COMPONENTS  *************************************** */
import NavBar from './dashboard/dashboardNavbar.jsx';
import TopContents from './recipeDetailsPage/TopContents.jsx';
import Ingredients from './recipeDetailsPage/Ingredients.jsx';
import Procedures from './recipeDetailsPage/Procedures.jsx';
import ReviewsForm from './recipeDetailsPage/ReviewsForm.jsx';
import Reviews from './recipeDetailsPage/Reviews.jsx';
import Footer from './footer.jsx';
/** ****************************** RECIPE DETAILS ACTIONS  ****************************************** */
import setCurrentRecipeRequest from '../actions/actionCreators/setCurrentRecipeActions';

class RecipeDetailsPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      topContents: {
        title: '', recipeImage: '', upvotes: 0, downvotes: 0
      },
      ingredients: '',
      procedures: ''
    };
  }

  componentDidMount() {
    const recipeId = this.props.match.params.id;
    this.props.setCurrentRecipeRequest(recipeId);
  }
  componentWillReceiveProps(nextProps) {
    const { title, ingredients, procedures, recipeImage, upvotes, downvotes } = nextProps.recipeDetails.recipe;
    if (nextProps.recipeDetails.recipe.title) {
      this.setState({
        topContents: { title, ingredients, procedures, recipeImage, upvotes, downvotes },
        ingredients,
        procedures
      });
    }
  }

  render() {
    return (
      <div>
        <div className="site-wrapper">
          <NavBar/>
          <div className="container">
            <main className="pl-4 pr-3 mt-4">
              <TopContents details={this.state.topContents}/>
              <div className="row mb-4">
                <Ingredients ingredients={this.state.ingredients}/>
                <div className="col-1"></div>
                <Procedures procedures={this.state.procedures}/>
              </div>
              <ReviewsForm/>
              <Reviews/>
            </main>
          </div>
        </div>
        <Footer id="homeFooter"/>
      </div>
    );
  }
}

RecipeDetailsPage.propTypes = {
  match: PropTypes.object.isRequired,
  setCurrentRecipeRequest: PropTypes.func.isRequired,
  recipeDetails: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
    recipeDetails: state.setCurrentRecipe.currentSetRecipe
  }
);

export default connect(mapStateToProps, { setCurrentRecipeRequest })(RecipeDetailsPage);
