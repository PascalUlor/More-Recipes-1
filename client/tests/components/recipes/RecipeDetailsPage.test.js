/* eslint-disable */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import { RecipeDetailsPage } from '../../../components/RecipeDetailsPage.jsx';


/**
 * @param { boolean } loading
 *
 * @return { * } null
 */
const setup = () => {
  const props = {
    setCurrentRecipe: jest.fn(),
    postReview: jest.fn(),
    addFavorite: jest.fn(),
    voteRecipe: jest.fn(),
    recipeDetails: {},
    addFavoriteSuccess: '',
    addFavoriteError: '',
    voteSuccessMessage: '',
    voteFailureMessage: '',
    match: {
      params: {
        id: 1
      }
    },
  };
  const store = mockStore({});
  return shallow((
    <Provider store={store}>
      <Router><RecipeDetailsPage {...props}/></Router>
    </Provider>
  ));
};

describe('RecipeDetailsPage component', () => {
  const mountWrapper = setup();
  it('should render correctly', () => {
    expect(mountWrapper).toMatchSnapshot();
  });
});
