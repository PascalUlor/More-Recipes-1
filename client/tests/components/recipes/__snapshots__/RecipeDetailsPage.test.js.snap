// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetailsPage component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <BrowserRouter>
      <RecipeDetailsPage
        addFavorite={[MockFunction]}
        addFavoriteError=""
        addFavoriteSuccess=""
        match={
          Object {
            "params": Object {
              "id": 1,
            },
          }
        }
        postReview={[MockFunction]}
        recipeDetails={Object {}}
        setCurrentRecipe={[MockFunction]}
        voteFailureMessage=""
        voteRecipe={[MockFunction]}
        voteSuccessMessage=""
      />
    </BrowserRouter>
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <RecipeDetailsPage
        addFavorite={[MockFunction]}
        addFavoriteError=""
        addFavoriteSuccess=""
        match={
          Object {
            "params": Object {
              "id": 1,
            },
          }
        }
        postReview={[MockFunction]}
        recipeDetails={Object {}}
        setCurrentRecipe={[MockFunction]}
        voteFailureMessage=""
        voteRecipe={[MockFunction]}
        voteSuccessMessage=""
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addFavorite": [MockFunction],
        "addFavoriteError": "",
        "addFavoriteSuccess": "",
        "match": Object {
          "params": Object {
            "id": 1,
          },
        },
        "postReview": [MockFunction],
        "recipeDetails": Object {},
        "setCurrentRecipe": [MockFunction],
        "voteFailureMessage": "",
        "voteRecipe": [MockFunction],
        "voteSuccessMessage": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <RecipeDetailsPage
          addFavorite={[MockFunction]}
          addFavoriteError=""
          addFavoriteSuccess=""
          match={
            Object {
              "params": Object {
                "id": 1,
              },
            }
          }
          postReview={[MockFunction]}
          recipeDetails={Object {}}
          setCurrentRecipe={[MockFunction]}
          voteFailureMessage=""
          voteRecipe={[MockFunction]}
          voteSuccessMessage=""
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addFavorite": [MockFunction],
          "addFavoriteError": "",
          "addFavoriteSuccess": "",
          "match": Object {
            "params": Object {
              "id": 1,
            },
          },
          "postReview": [MockFunction],
          "recipeDetails": Object {},
          "setCurrentRecipe": [MockFunction],
          "voteFailureMessage": "",
          "voteRecipe": [MockFunction],
          "voteSuccessMessage": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
